### ISSUE TITLE
Assistant runs after `setup-alire@latest-devel` action despite toolchain being already configured

### ISSUE THREAD

**Issue by onox** [alire-project/alire#851]
In a GitHub Actions yml file, after running ada-actions/toolchain@ce2021 and alire-project/setup-alire@latest-devel on the ubuntu-latest OS, I would expect that `alr build` by default uses the gnat_external toolchain if it detects its presence on the system. Instead `alr build` fails with the following error:

```
Run alr build
Welcome to the toolchain selection assistant

In this assistant you can set up the default toolchain to be used with any crate
that does not specify its own top-level dependency on a version of gnat or 
gprbuild.

If you choose "None", Alire will use whatever version is found in the 
environment.

Note: Currently configured: gnat_native=11.2.1

Please select the gnat version for use with this configuration
  1. gnat_native=11.2.1
  2. None
  3. gnat_external=9.3.0 [Detected at /usr/bin/gnat]
  4. gnat_arm_elf=11.2.1
  5. gnat_riscv64_elf=11.2.1
  6. gnat_arm_elf=10.3.1
  7. gnat_native=10.3.1
  8. gnat_riscv64_elf=10.3.1
Enter your choice index (first is default): 
> ERROR: Canceled.
Error: Process completed with exit code 1.
```

Of course a user can/may switch to another toolchain, but if the user has run `apt install gnat` (as shown by `[Detected at /usr/bin/gnat]`), isn't that an indication that the user wants to use the toolchain from the system package repository (apt in this case)?

**Comment by mosteo** [comment]
Thanks for the report, @onox. There are several issues here:

- By default, `setup-alire@latest-devel` will configure the native packaged toolchain
   - This can be overridden with the `toolchain` argument for the action
- No matter an external being detected, the assistant always defaults to the native packaged toolchain.
   - This was a decision we made, to always have the same default.
- The action configures the toolchain, hence it should not run when you run `alr build`, but:
   - There is a bug in the devel branch or the action that causes the assistant to re-run. (This is the only real issue to fix I can see right now).

TL;DR: your best option at this time is to pass the explicit toolchain you want to use to the action, and if the assistant kicks in anyway, configure it again with `alr toolchain --select` as the first thing in your workflow, until we get that fixed.

**Comment by mosteo** [comment]
Our setup actions are working as intended, AFAIK, after https://github.com/alire-project/setup-alire/pull/24 and https://github.com/alire-project/setup-alire/pull/25/files. Please reopen if you detect any new issues.
